app/components/date-time-picker.js

```diff
@@ -1,22 +1,29 @@
 import Component from '@glimmer/component';
 import { action } from "@ember/object";
 import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
 
 export default class DateTimePicker extends Component{
   @service intl;
-  date;
-  hours;
-  minutes;
+  @tracked date;
 
   constructor() {
     super(...arguments);
     if(this.args.value) {
       this.date = new Date(this.args.value);
-      this.hours = this.date.getHours();
-      this.minutes = this.date.getMinutes();
+    } else {
+      this.date = new Date();
     }
   }
 
+  get hours() {
+    return this.date.getHours();
```

Comment:
This uses (abuses?) the fact that getHours will change the date when the value goes "out of bounds". 

- 24 -> date + 1, hour to 0
- -1 -> date - 1, hour to 23